pennylane/templates/state_preparations/cosine_window.py

```diff
@@ -0,0 +1,129 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Contains the CosineWindow template.
+"""
+import numpy as np
+import pennylane as qml
+from pennylane.operation import AnyWires, StatePrepBase
+from pennylane import math
+from pennylane.wires import Wires, WireError
+
+
+class CosineWindow(StatePrepBase):
+    r"""CosineWindow(wires)
```

Comment:
Is there a paper we should reference ?

Reply:
I am going to reference one that I have based on, although I had to make some home-made adjustments to make it make sense in our framework